core/src/test/scala/unit/kafka/admin/DeleteTopicTest.scala

```diff
@@ -187,264 +220,337 @@ class DeleteTopicTest extends QuorumTestHarness {
+    val follower = brokersWithReplicas.filter(s => s.config.brokerId != leaderNode.id()).last
     follower.shutdown()
     // start topic deletion
-    adminZkClient.deleteTopic(topic)
+    adminClient.deleteTopics(Collections.singleton(topic))
 
     // make sure deletion of all of the topic's replicas have been tried
-    ensureControllerExists()
-    val (controller, controllerId) = getController()
-    val allReplicasForTopic = getAllReplicasFromAssignment(topic, expectedReplicaAssignment)
-    TestUtils.waitUntilTrue(() => {
-      val replicasInDeletionSuccessful = controller.kafkaController.controllerContext.replicasInState(topic, ReplicaDeletionSuccessful)
-      val offlineReplicas = controller.kafkaController.controllerContext.replicasInState(topic, OfflineReplica)
-      allReplicasForTopic == (replicasInDeletionSuccessful union offlineReplicas)
-    }, s"Not all replicas for topic $topic are in states of either ReplicaDeletionSuccessful or OfflineReplica")
-
+    if (isKRaftTest()) {
+      val aliveBrokers = brokers.filter(b => b != follower)
+      TestUtils.waitForAllPartitionsMetadata(aliveBrokers, topic, 0)
+      TestUtils.waitForAllPartitionsMetadata(Seq(follower), topic, 1)
+    } else {
+      ensureControllerExists()
+      val (controller, _) = getController()
+      val allReplicasForTopic = getAllReplicasFromAssignment(topic, expectedReplicaAssignment)
+      TestUtils.waitUntilTrue(() => {
+        val replicasInDeletionSuccessful = controller.kafkaController.controllerContext.replicasInState(topic, ReplicaDeletionSuccessful)
+        val offlineReplicas = controller.kafkaController.controllerContext.replicasInState(topic, OfflineReplica)
+        allReplicasForTopic == (replicasInDeletionSuccessful union offlineReplicas)
+      }, s"Not all replicas for topic $topic are in states of either ReplicaDeletionSuccessful or OfflineReplica")
+    }
     // increase the partition count for topic
-    val props = new Properties()
-    props.setProperty(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, TestUtils.plaintextBootstrapServers(servers))
-    val adminClient = Admin.create(props)
     try {
       adminClient.createPartitions(Map(topic -> NewPartitions.increaseTo(2)).asJava).all().get()
     } catch {
       case _: ExecutionException =>
     }
     // trigger a controller switch now
-    val previousControllerId = controllerId
-
-    controller.shutdown()
-
-    ensureControllerExists()
-    // wait until a new controller to show up
-    TestUtils.waitUntilTrue(() => {
-      val (newController, newControllerId) = getController()
-      newControllerId != previousControllerId
-    }, "The new controller should not have the failed controller id")
-
-    // bring back the failed brokers
-    follower.startup()
-    controller.startup()
-    TestUtils.verifyTopicDeletion(zkClient, topic, 2, servers)
-    adminClient.close()
+    if (isKRaftTest()) {
```

Comment:
I would consider writing two different tests instead of using this pattern so often.